server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -414,6 +423,75 @@ private void createMetadataBasedIterables(Metadata metadata) {
         routines = () -> sequentialStream(routineInfos).filter(Objects::nonNull).iterator();
     }
 
+    public Set<String> enabledRoles(Role role, Roles roles) {
+        Set<String> enabledRoles = new HashSet<>();
+        enabledRoles.add(role.name());
+        for (GrantedRole grantedRole: role.grantedRoles()) {
+            enabledRoles.add(grantedRole.roleName());
+            Role retrievedRole = roles.findRole(grantedRole.roleName());
+            if (retrievedRole != null) {
+                if (!retrievedRole.grantedRoles().isEmpty()) {
+                    enabledRoles.addAll(enabledRoles(retrievedRole, roles));
+                }
+            }
+        }
+        return enabledRoles;
+    }
+
+    public Iterable<ApplicableRole> applicableRoles(Role role, Roles roles) {
+        List<ApplicableRole> applicableRoles = new ArrayList<>();
+
+        for (GrantedRole grantedRole: role.grantedRoles()) {
+            ApplicableRole newRole = new ApplicableRole();
+            newRole.setGrantee(role.name());
+            newRole.setRoleName(grantedRole.roleName());
+            newRole.setGrantable(roles.hasPrivilege(role, Permission.AL, Securable.CLUSTER, null));
+            applicableRoles.add(newRole);
+        }
+
+        return applicableRoles;
+    }
+
+    public Iterable<RoleTableGrant> roleTableGrants(Role role, Roles roles) {
+        List<RoleTableGrant> roleTableGrants = new ArrayList<>();
+        role.privileges().forEach(p -> {
```

Comment:
`roles.privileges()`, only shows the direct privileges granted to a user, but I think we need to check the whole hierarchy (privileges inherited due to granted roles). How does postgres behave here? If we need to check the hierarchy, then we need to iterate over all tables, and use the `roles.hasPrivilege()` on each table to get the privilege (possible inherited) that applies.